contrib/hbcurl/core.c

```diff
@@ -1034,6 +1051,50 @@ HB_FUNC( CURL_EASY_SETOPT )
                }
             }
             break;
+            case HB_CURLOPT_HTTPPOST_FORM:
+            {
+               PHB_ITEM pArray = hb_param( 3, HB_IT_ARRAY );
+
+               if( pArray )
+               {
+                  HB_SIZE ulPos;
+                  HB_SIZE ulArrayLen = hb_arrayLen( pArray );
+
+                  for( ulPos = 0; ulPos < ulArrayLen; ++ulPos )
+                  {
+                     PHB_ITEM pSubArray = hb_arrayGetItemPtr( pArray, ulPos + 1 );
+
+                     int type = hb_arrayGetNI( pSubArray, 1 );
+
+                     switch( type )
+                     {
+                        case 1:
+
+                           curl_formadd( &hb_curl->pHTTPPOST_First,
+                                         &hb_curl->pHTTPPOST_Last,
+                                         CURLFORM_COPYNAME, hb_arrayGetCPtr( pSubArray, 2 ),
+                                         CURLFORM_NAMELENGTH, hb_arrayGetCLen( pSubArray, 2 ),
+                                         CURLFORM_COPYCONTENTS, hb_arrayGetCPtr( pSubArray, 3 ),
+                                         CURLFORM_CONTENTSLENGTH, hb_arrayGetCLen( pSubArray, 3 ),
+                                         CURLFORM_END );
+
+                           break;
+                        case 2:
```

Comment:
// HB_CURLOPT_HTTPPOST_FORM

No, it actually generates a slightly different  POST.
This is an extract of the working code I'm using to show the issue:

#define HTTP_FORM_TYPE_INPUT	1
#define HTTP_FORM_TYPE_FILE		2

aForm :=  {}

AADD( aForm, {HTTP_FORM_TYPE_INPUT, "corporationCode", corp})	// Corporation Code 
AADD( aForm, {HTTP_FORM_TYPE_INPUT, "shipToCode", shipTo})		 // Ship to Code 
AADD( aForm, {HTTP_FORM_TYPE_INPUT, "valueCode", rne }) // RNE Number	
AADD( aForm, {HTTP_FORM_TYPE_FILE,  "fileUpload", fullPath})		  // Selected File
AADD( aForm, {HTTP_FORM_TYPE_INPUT, "fileName", fileName})			 // Selected File Name

curl_easy_setopt( ::pCurl, HB_CURLOPT_HTTPPOST_FORM, aForm )

This is the post taken with a sniffer:

Content-Type: multipart/form-data; boundary=------------------------ff836f997cccf49c
--------------------------ff836f997cccf49c
Content-Disposition: form-data; name="corporationCode"
LGEIS
--------------------------ff836f997cccf49c
Content-Disposition: form-data; name="shipToCode"
70062202
--------------------------ff836f997cccf49c
Content-Disposition: form-data; name="valueCode"
RNE150615063267
--------------------------ff836f997cccf49c
Content-Disposition: form-data; name="fileUpload"; filename="RNE150615063267.jpg"
Content-Type: image/jpeg

------------------------------------------------
Only the fileUpload is actually a file. the other fields are in memory strings.